File: result/translate/starchat/java2cpp/c++/NEWMAN_CONWAY_SEQUENCE_1.cpp
int f_gold ( int n ) {
    vector <int> f(n+1);
    f[0] = 0;
    f[1] = 1;
    f[2] = 1;
    for (int i = 3; i <= n; i++) f[i] = f[i-1] + f[i-f[i-1]];
    return f[n];
}

File: result/translate/transcoder/python2java/java/NEWMAN_CONWAY_SEQUENCE_1.java
static int f_gold ( int n ) {
  int [ ] f = new int [ n + 1 ] ;
  for ( int i = 3 ;
  i <= n ;
  i ++ ) {
    int r = f [ f [ i - 1 ] ] + f [ i - f [ i - 1 ] ] ;
    f [ i ] = r ;
  }
  return f [ n ] ;
}